app/src/main/java/com/zulip/android/activities/PhotoSendActivity.java

```diff
@@ -0,0 +1,142 @@
+
+        setContentView(R.layout.activity_photo_send);
+
+        // get the file path sent from ZulipActivity
+        final Intent intent = getIntent();
+        mPhotoPath = intent.getStringExtra(Intent.EXTRA_TEXT);
+
+        mImageView = (ImageView) findViewById(R.id.photoImageView);
+        ImageView sendPhoto = (ImageView) findViewById(R.id.send_photo);
+
+        // intent to go back to ZulipActivity and upload photo
+        final Intent sendIntent = new Intent(this, ZulipActivity.class);
+        sendIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+
+        sendPhoto.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (mIsCropped) {
+                    // if image was cropped, delete old file
+                    // and store new bitmap on that location
+                    Bitmap bitmap = ((BitmapDrawable) mImageView.getDrawable()).getBitmap();
+                    PhotoHelper.saveBitmapAsFile(mPhotoPath, bitmap);
+                }
+
+                // add the file path of cropped image
+                sendIntent.putExtra(Intent.EXTRA_TEXT, mPhotoPath);
+                startActivity(sendIntent);
+            }
+        });
+
+        ImageView deleteBtn = (ImageView) findViewById(R.id.delete_photo);
+        deleteBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                File file = new File(mPhotoPath);
+                boolean isFileDeleted = file.delete();
+                if (!isFileDeleted) {
+                    Log.e("Photo upload", "Could not delete photo");
+                }
+                // go back to ZulipActivity to start camera intent
+                startActivity(sendIntent);
+            }
+        });
+
+        ImageView editPhotoBtn = (ImageView) findViewById(R.id.edit_photo);
+        editPhotoBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (mIsCropped) {
+                    // if image was cropped, delete old file
+                    // and store new bitmap on that location
+                    Bitmap bitmap = ((BitmapDrawable) mImageView.getDrawable()).getBitmap();
+                    PhotoHelper.saveBitmapAsFile(mPhotoPath, bitmap);
+                }
+
+                // start PhotoEditActivity, passing it the file path for cropped photo
+                Intent intent = new Intent(PhotoSendActivity.this, PhotoEditActivity.class);
+                intent.putExtra(Intent.EXTRA_TEXT, mPhotoPath);
+                intent.putExtra(mIsCroppedKey, mIsCropped);
+                startActivity(intent);
+            }
+        });
+
+        mCropImageView = (CropImageView) findViewById(R.id.crop_image_view);
+
+        final ImageView cropBtn = (ImageView) findViewById(R.id.crop_btn);
+        cropBtn.setOnClickListener(new View.OnClickListener() {
```

Comment:
@saketkumar95 Thanks for pointing this point. 
After looking at how other successful apps deal with this, I think user should return to the photo capturing screen (sending intent again). @kunall17 What do you think?